File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i in range(len(nums)):
            match = target - nums[i]
            if match in dict:
                return [i, dict[match]]
            dict[nums[i]] = i
            
        return []
        

File: 1-two-sum/1-two-sum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i=0; i < nums.length; i++) {
            int match = target - nums[i];
            if (map.containsKey(match)) {
                return new int[] {map.get(match), i};
            }
            else {
                map.put(nums[i], i);
            }
            
        }
        return null;
     
    }
}